services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_utils.py

```diff
@@ -442,3 +451,24 @@ async def prepare_services_environment(
     )
 
     scheduler_data.dynamic_sidecar.is_service_environment_ready = True
+
+
+async def get_allow_metrics_collection(
+    app: FastAPI, user_id: UserID, product_name: ProductName
+) -> bool:
+    repo = get_repository(app, UserPreferencesFrontendRepository)
+    preference: FrontendUserPreference | None = await repo.get_user_preference(
+        user_id=user_id,
+        product_name=product_name,
+        preference_class=AllowMetricsCollectionFrontendUserPreference,
+    )
+
+    if preference is None:
+        return cast(
+            bool, AllowMetricsCollectionFrontendUserPreference.get_default_value()
+        )
+
+    allow_metrics_collection = AllowMetricsCollectionFrontendUserPreference.parse_obj(
```

Comment:
Q: why not have a general mechanism to resolve the settings you are looking for. If I understnd this right, the front-end settings in the database are kind of a blob. Then you defines some "fields " of interest for the backend by creating a class e.g. `AllowMetricsCollectionFrontendUserPreference`

I assume most of the front-end settings concern the front-end i.e. does not need this class but this class is used kind of to "deserialized" what might be of used for the backend.  I wonder if we could have a generic function like
```python
allow_mettrics, other, another = get_preferences_with(select=[AllowMetricsCollectionFrontendUserPreference, OtherPreference, AnotherPreference])
```

To me it resembles a lot sqlalchemy queries where you can get from the database the columns you like and the client creates a proxyRow object with the corresponding attribbutes.

Reply:
There was no need for something more advanced. Since it was only used n one place.
I will refactor it to something like you propose if this is required again. For now we keep it simple.

Reply:
i find the current approach very error prone and therefore not maintainable ... but I guess it is ok for now :-)